core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,929 @@ def update_question_suggestion(
+
+    Returns:
+        list(TranslationReviewStats). TranslationReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    translation_review_stats_models = (
+        suggestion_models.TranslationReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_translation_review_stats_from_model(model)
+        for model in translation_review_stats_models
+    ]
+
+
+def _get_all_question_contribution_stats(
+    user_id: str
+) -> List[suggestion_registry.QuestionContributionStats]:
+    """Gets all QuestionContributionStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionContributionStats). QuestionContributionStats domain
+        objects corresponding to the supplied user.
+    """
+    question_contribution_stats_models = (
+        suggestion_models.QuestionContributionStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_contribution_stats_from_model(model)
+        for model in question_contribution_stats_models
+    ]
+
+
+def _get_all_question_review_stats(
+    user_id: str
+) -> List[suggestion_registry.QuestionReviewStats]:
+    """Gets all QuestionReviewStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionReviewStats). QuestionReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    question_review_stats_models = (
+        suggestion_models.QuestionReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_review_stats_from_model(model)
+        for model in question_review_stats_models
+    ]
+
+
+def get_all_contributor_stats(
+    user_id: str
+) -> suggestion_registry.ContributorStatsSummary:
+    """Gets ContributorStatsSummary corresponding to the supplied user.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        ContributorStatsSummary. ContributorStatsSummary domain objects
+        corresponding to the supplied user.
+    """
+    translation_contribution_stats = get_all_translation_contribution_stats(
+        user_id)
+    translation_review_stats = _get_all_translation_review_stats(user_id)
+    question_contribution_stats = _get_all_question_contribution_stats(user_id)
+    question_review_stats = _get_all_question_review_stats(user_id)
+
+    return suggestion_registry.ContributorStatsSummary(
+        user_id,
+        translation_contribution_stats,
+        question_contribution_stats,
+        translation_review_stats,
+        question_review_stats)
+
+
+def _update_translation_contribution_stats_models(
+    translation_contribution_stats: List[
+        suggestion_registry.TranslationContributionStats]
```

Comment:
Why does this function need to accept a list? Don't the callers currently always only supply a single stats object?

Reply:
I thought in future if we need to update a list of stats data, it would be helpful so I kept it as a list.